src/encoded/commands/deploy.py

```diff
@@ -200,37 +217,69 @@ def create_ec2_instances(client, image_id, count, instance_type, security_groups
     )
     return reservations
 
+
 def tag_ec2_instance(instance, name, branch, commit, username, elasticsearch, cluster_name):
-    tags=[
+    tags = [
         {'Key': 'Name', 'Value': name},
         {'Key': 'branch', 'Value': branch},
         {'Key': 'commit', 'Value': commit},
         {'Key': 'started_by', 'Value': username},
     ]
     if elasticsearch == 'yes':
         tags.append({'Key': 'elasticsearch', 'Value': elasticsearch})
-    if cluster_name != None:
+    if cluster_name is not None:
         tags.append({'Key': 'ec_cluster_name', 'Value': cluster_name})
     instance.create_tags(Tags=tags)
     return instance
 
+
 def tag_spot_instance(instance, name, branch, commit, username, elasticsearch, client, cluster_name):
-    tags=[
+    tags = [
         {'Key': 'Name', 'Value': name},
         {'Key': 'branch', 'Value': branch},
         {'Key': 'commit', 'Value': commit},
         {'Key': 'started_by', 'Value': username},
     ]
     if elasticsearch == 'yes':
         tags.append({'Key': 'elasticsearch', 'Value': elasticsearch})
-    if not cluster_name == None:
+    if cluster_name is not None:
         tags.append({'Key': 'ec_cluster_name', 'Value': cluster_name})
     instance_id = client.create_tags(Resources=[get_instance_id(instance, client)], Tags=tags)
     return instance_id
 
-def run(wale_s3_prefix, image_id, instance_type, elasticsearch, spot_instance, spot_price, cluster_size, cluster_name, check_price,
-        branch=None, name=None, role='demo', profile_name=None, teardown_cluster=None):
-    
+
+def read_ssh_key():
+    home = expanduser("~")
+    ssh_key_path = home + '/' + '.ssh/id_rsa.pub'
+    ssh_keygen_args = ['ssh-keygen', '-l', '-f', ssh_key_path]
+    fingerprint = subprocess.check_output(
+        ssh_keygen_args
+    ).decode('utf-8').strip()
+    if fingerprint:
+        with open(ssh_key_path, 'r') as f:
+            ssh_pub_key = f.readline().strip()
+            return ssh_pub_key
+
+
+def get_user_data(commit, config_file, data_insert):
+    cmd_list = ['git', 'show', commit + config_file]
+    config_template = subprocess.check_output(cmd_list).decode('utf-8')
+    ssh_pub_key = read_ssh_key()
+    if not ssh_pub_key:
+        # We can block deploy or add empty line??
```

Comment:
There is no reason in principle why someone should not be able to deploy a (demo) instance without a key.
So, I think it should just maybe print a warning:  "WARNING:  User is not authorized with ssh access to new instance because they have no ssh key"